home-manager.nix

```diff
@@ -73,28 +73,29 @@ in
                 dirListToPath (tail (splitPath [ dir ]))
               else
                 dir;
-            targetDir = concatPaths [ persistentStoragePath dir ];
-            mountPoint = concatPaths [ config.home.homeDirectory mountDir ];
+            targetDir = escapeShellArg (concatPaths [ persistentStoragePath dir ]);
+            mountPoint = escapeShellArg (concatPaths [ config.home.homeDirectory mountDir ]);
             name = "bindMount-${sanitizeName targetDir}";
             startScript = pkgs.writeShellScript name ''
               set -eu
-              if ! ${pkgs.utillinux}/bin/mount | ${pkgs.gnugrep}/bin/grep -E '${mountPoint}( |/)'; then
-                  ${pkgs.bindfs}/bin/bindfs -f --no-allow-other "${targetDir}" "${mountPoint}"
+              if ! ${pkgs.utillinux}/bin/mount | ${pkgs.gnugrep}/bin/grep -F ${mountPoint}' ' \
+                 && ! ${pkgs.utillinux}/bin/mount | ${pkgs.gnugrep}/bin/grep -F ${mountPoint}/; then
+                  ${pkgs.bindfs}/bin/bindfs -f --no-allow-other ${targetDir} ${mountPoint}
               else
-                  echo "There is already an active mount at or below ${mountPoint}!" >&2
+                  echo "There is already an active mount at or below "${mountPoint}"!" >&2
```

Comment:
Maybe it's just me, but I think rendering the single quotes is desirable (e.g. drop the added double quotes). Take, for example: a path with a space in it, `/root/some/mount point`. With these double quotes, the message will display as:

```
There is already an active mount at or below /root/some/mount point!
```

But without these double quotes, the message will display as:

```
There is already an active mount at or below '/root/some/mount point'!
```

Reply:
Yep, that looks better; I'll drop the added double quotes.